src/freehand_selection.jl

```diff
@@ -0,0 +1,60 @@
+function init_freehand_select(ctx::ImageContext)
+    c = ctx.canvas
+    enabled = Signal(true)
+    dragging = Signal(false)
+
+    dummybtn = MouseButton{UserUnit}()
+
+    min_x = Signal(1.0)
+    max_x = Signal(Float64(size(ctx.image,2)))
+    min_y = Signal(1.0)
+    max_y = Signal(Float64(size(ctx.image,1)))
+
+    sigstart = map(filterwhen(enabled, dummybtn, c.mouse.buttonpress)) do btn
+        if !ispolygon(value(ctx.points)) || !isinside(Point(btn.position), Point.(value(ctx.points)))
+            push!(dragging, true)
+            push!(ctx.shape, Rectangle()) # some identifier of type of selection
```

Comment:
I almost think it would be better to create an `UndefinedShape()` type just so you're not "lying".